tools/scripts/upload-to-nr.js

```diff
@@ -33,137 +34,186 @@ var argv = yargs
+      }
     }
-  }
 
-  var envOptions = {
-    staging: {
-      url: 'https://staging-api.newrelic.com/v2/js_agent_loaders/create.json',
-      headers: {
-        'X-Api-Key': argv['staging-api-key']
-      }
-    },
-    eu: {
-      url: 'https://api.eu.newrelic.com/v2/js_agent_loaders/create.json',
-      headers: {
-        'X-Api-Key': argv['eu-api-key']
-      }
-    },
-    production: {
-      url: 'https://api.newrelic.com/v2/js_agent_loaders/create.json',
-      headers: {
-        'X-Api-Key': argv['production-api-key']
+    var envOptions = {
+      staging: {
+        url: 'https://staging-api.newrelic.com/v2/js_agent_loaders/create.json',
+        headers: {
+          'X-Api-Key': argv['staging-api-key']
+        }
+      },
+      eu: {
+        url: 'https://api.eu.newrelic.com/v2/js_agent_loaders/create.json',
+        headers: {
+          'X-Api-Key': argv['eu-api-key']
+        }
+      },
+      production: {
+        url: 'https://api.newrelic.com/v2/js_agent_loaders/create.json',
+        headers: {
+          'X-Api-Key': argv['production-api-key']
+        }
       }
     }
-  }
 
-  console.log('Uploading loader ' + filename + ' to ' + environment + '...')
-  var options = util._extend(envOptions[environment], baseOptions)
+    console.log('Uploading loader ' + filename + ' to ' + environment + '...')
+    var options = util._extend(envOptions[environment], baseOptions)
 
-  request(options, function (err, res, body) {
-    if (err) return cb(err)
-    if (res.statusCode === 200) {
-      console.log('Uploaded loader version ' + filename + ' to ' + environment)
-      return cb()
-    }
+    request(options, function (err, res, body) {
+      if (err) return cb(err)
+      if (res.statusCode === 200) {
+        console.log('Uploaded loader version ' + filename + ' to ' + environment)
+        return cb()
+      }
 
-    cb(new Error('Failed to upload ' + filename + ' loader to ' + environment + ' db: (' + res.statusCode + ') ' + JSON.stringify(body)))
-  })
-}
+      cb(new Error('Failed to upload ' + filename + ' loader to ' + environment + ' db: (' + res.statusCode + ') ' + JSON.stringify(body)))
+    })
+  }
 
-function loaderFilenames() {
-  const buildDir = path.resolve(__dirname, '../../build/')
-  return fs.readdirSync(buildDir).filter(x => x.startsWith('nr-loader') && x.endsWith('.js')) 
-}
+  /**
+   * Fetches an array of loader filenames and contents from the CDN
+   * @returns {Promise<{[fileName]: string}[]>} Promise contains an array of objects {[filename]: body} --> {'nr-loader-spa-1221.min.js': ...scriptContents}
+   */
+  async function loaderFilenames() {
+    const loaderTypes = ['rum', 'full', 'spa']
+    const version = argv['version']
+    const fileNames = loaderTypes.map(type => [
+      `nr-loader-${type}-${version}.min.js`, 
+      `nr-loader-${type}-polyfills-${version}.min.js`,
+      `nr-loader-${type}-${version}.js`, 
+      `nr-loader-${type}-polyfills-${version}.js`,
```

Comment:
[question] Is there ever a chance that NR1 uses the non-minified loader? if not, does it make sense to not upload it?

Reply:
Not sure on that one.  I know there are a few flags in the connect service for "dev" builds, which *i think* signal it to return the unminified version.  Not sure if that is still used